packages/fluentui/react-northstar/src/components/Icon/Icon.tsx

```diff
@@ -147,7 +161,73 @@ Icon.propTypes = {
 };
 Icon.handledProps = Object.keys(Icon.propTypes) as any;
 
-Icon.create = createShorthandFactory({ Component: Icon, mappedProp: 'name', allowsJSX: false });
+/**
+ * This augmented version of the create method for the Icon should help
+ * with the migration path from the old to the new icons definition.
+ */
+const inconsistentIconNames: Record<string, string> = {
+  AcceptIcon: 'icon-checkmark',
+  ChevronDownMediumIcon: 'icon-menu-arrow-down',
+  ChevronEndMediumIcon: 'icon-menu-arrow-end',
+  CircleIcon: 'icon-circle',
+  CloseIcon: 'icon-close',
+  OneDriveIcon: 'onedrive',
+  OneNoteColorIcon: 'onenote-color',
+  OneNoteIcon: 'onenote',
+  PauseIcon: 'icon-pause',
+  PlayIcon: 'icon-play',
+  PowerPointColorIcon: 'powerpoint-color',
+  PowerPointIcon: 'powerpoint',
+  TriangleDownIcon: 'icon-arrow-down',
+  TriangleEndIcon: 'icon-arrow-end',
+  TriangleUpIcon: 'icon-arrow-up',
+};
+
+function isFluentJSXIcon(value: ShorthandValue<IconProps>): value is React.ReactElement<IconProps, typeof Icon> {
+  return (
+    React.isValidElement(value) &&
+    typeof value.type === 'function' &&
+    (value.type as React.ComponentType).displayName?.indexOf('Icon') !== -1
+  );
+}
+
+function toIconNameFromDisplayName(displayName: string) {
+  return inconsistentIconNames[displayName] ? inconsistentIconNames : _.kebabCase(displayName.replace('Icon', ''));
+}
+
+const iconOriginalShorthandFactory = createShorthandFactory({
+  Component: Icon,
+  mappedProp: 'name',
+  allowsJSX: false,
+});
+
+const iconPatchedShorthandFactory: ShorthandFactory<IconProps> = (value, options) => {
+  if (process.env.NODE_ENV === 'development') {
+    if (typeof value === 'string') {
+      console.warn(
+        `Fluent UI: Deprecation notice, please use JSX icon as a value instead of a string, for example replace <Button icon="bell" /> with <Button icon={<BellIcon />} />.`,
+      );
+    }
+
+    if (_.isPlainObject(value)) {
+      if ((value as IconProps).name) {
+        console.warn(
+          `Fluent UI: Deprecation notice, please use JSX icon as a value instead of an object, for example replace <Button icon={{ name: 'bell', outline: true }} /> with <Button icon={<BellIcon outline />} />.`,
+        );
+      }
+    }
+  }
+
+  if (isFluentJSXIcon(value)) {
+    const iconName = toIconNameFromDisplayName(value.type.displayName.replace('Icon', ''));
```

Comment:
If `value.type.displayName` is `MoreIcon` it will cut it to `More` and then `inconsistentIconNames` will not work

```suggestion
    const iconName = toIconNameFromDisplayName(value.type.displayName);
```

Reply:
Good catch, we have the replace again here https://github.com/microsoft/fluentui/pull/14404/files#diff-b0039d331230fd1622db7d2eea352483R195